src/components/Cluster/ClusterDetail/NodePool.tsx

```diff
@@ -52,41 +53,108 @@ const NameWrapperDiv = styled.div`
   }
 `;
 
-class NodePool extends Component {
-  state = {
+interface INPViewAndEditName {
+  activateEditMode: () => boolean;
+  name: string;
+}
+
+type TNPViewAndEditName = React.Component<INPViewAndEditName>;
+
+interface IStateProps {
+  availableZonesGridTemplateAreas: string | undefined;
+  cluster: object;
+  nodePool: INodePool;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  nodePoolActions: Record<string, (...args: any[]) => Promise<any>>;
+  provider: string;
+}
+
+interface INodePoolsState {
+  isNameBeingEdited: boolean;
+}
+
+interface INodePoolProps extends IStateProps, IDispatchProps {}
+
+interface IScaleNodePoolModal {
+  reset: () => void;
+  show: () => void;
+  setNodePool: (nodePool: INodePool) => void;
+}
+
+class NodePool extends Component<INodePoolProps, INodePoolsState> {
+  static propTypes: IStateProps & IDispatchProps = {
+    /**
+     * We skip typechecking because we don't want to define the whole object
+     * structure (for now)
+     */
+    // @ts-ignore
+    availableZonesGridTemplateAreas: PropTypes.string,
+    cluster: PropTypes.object,
+    // @ts-ignore
+    nodePool: PropTypes.shape({
+      availability_zones: PropTypes.any, // TODO fix it.
+      id: PropTypes.string,
+      name: PropTypes.string,
+      node_spec: PropTypes.object,
+      scaling: PropTypes.shape({
+        min: PropTypes.number,
+        max: PropTypes.number,
+      }),
+      status: PropTypes.shape({
+        nodes: PropTypes.number,
+        nodes_ready: PropTypes.number,
+      }),
+    }),
+    // @ts-ignore
+    nodePoolActions: PropTypes.object,
+    // @ts-ignore
+    dispatch: PropTypes.func,
+    // @ts-ignore
+    provider: PropTypes.string,
+  };
+
+  public state: INodePoolsState = {
     isNameBeingEdited: false,
   };
 
-  toggleEditingState = (isNameBeingEdited) => {
+  private viewEditNameRef: React.RefObject<TNPViewAndEditName> | null = null;
+  private scaleNodePoolModal: React.RefObject<
```

Comment:
These aren't actually of type `RefObject`. `RefObject` is the type of ref we use with `ref.current`.
These are component refs, which make the class instance of the component be accessible from the parent component. That's why we can use `this.viewEditNameRef.activateEditMode()`.

Having these as `TNPViewAndEditName | null` would be more suitable